waspc/cli/src/Wasp/Cli/Command/CreateNewProject/ProjectDescription.hs

```diff
@@ -0,0 +1,104 @@
+    throwProjectCreationError,
+  )
+import Wasp.Cli.Command.CreateNewProject.StarterTemplates.Common
+  ( StarterTemplateName,
+    findTemplateNameByString,
+  )
+import qualified Wasp.Cli.Interactive as Interactive
+import Wasp.Project (WaspProjectDir)
+import Wasp.Util (indent, kebabToCamelCase)
+
+data NewProjectDescription = NewProjectDescription
+  { _projectName :: String,
+    _appName :: String,
+    _templateName :: StarterTemplateName,
+    _absWaspProjectDir :: Path' Abs (Dir WaspProjectDir)
+  }
+
+createNewProjectDescription :: NewProjectArgs -> [StarterTemplateName] -> Command NewProjectDescription
+createNewProjectDescription (NewProjectArgs projectNameArg templateNameArg) starterTemplateNames =
+  do
+    projectName <- getOrAskProjectName projectNameArg
+    absWaspProjectDir <- validateAndGetAbsProjectDir projectName
+
+    selectedTemplate <- selectTemplate starterTemplateNames
+    mkNewProjectDescription projectName absWaspProjectDir selectedTemplate
+  where
+    getOrAskProjectName :: Maybe String -> Command String
+    getOrAskProjectName = \case
+      Just projectName -> return projectName
+      Nothing -> liftIO $ Interactive.askForRequiredInput "Enter the project name (e.g. my-project)"
+
+    validateAndGetAbsProjectDir :: String -> Command (Path' Abs (Dir WaspProjectDir))
+    validateAndGetAbsProjectDir projectName = do
+      absWaspProjectDir <- getAbsPathToNewProjectDirInCwd projectName
+      ensureProjectDirDoesNotExist projectName absWaspProjectDir
+      return absWaspProjectDir
+      where
+        ensureProjectDirDoesNotExist :: String -> Path' Abs (Dir WaspProjectDir) -> Command ()
+        ensureProjectDirDoesNotExist projectDirName absWaspProjectDir = do
+          dirExists <- doesDirExist $ toPathAbsDir absWaspProjectDir
+
+          when dirExists $
+            throwProjectCreationError $
+              "Directory \"" ++ projectDirName ++ "\" is not empty."
+
+    selectTemplate :: [StarterTemplateName] -> Command StarterTemplateName
+    selectTemplate templateNames = do
+      let maybeTemplateName = templateNameArg >>= findTemplateNameByString templateNames
+      throwIfUserProvidedInvalidTemplateNameViaArgs maybeTemplateName
+      getOrAskTemplateName maybeTemplateName
+      where
+        throwIfUserProvidedInvalidTemplateNameViaArgs :: Maybe StarterTemplateName -> Command ()
+        throwIfUserProvidedInvalidTemplateNameViaArgs maybeTemplateName = do
+          let isTemplateNameArgProvided = isJust templateNameArg
+              isTemplateNameFound = isJust maybeTemplateName
+          when
+            (isTemplateNameArgProvided && not isTemplateNameFound)
+            throwInvalidTemplateNameUsedError
+
+        getOrAskTemplateName :: Maybe StarterTemplateName -> Command StarterTemplateName
+        getOrAskTemplateName = \case
+          Just templateName -> return templateName
+          Nothing -> liftIO $ Interactive.askToChoose "Choose a starter template" templateNames
```

Comment:
We don't have to do it for this round, but I'm just wondering if we are in non-interactive mode and they don't specify a template, should we just use the default? That would allow any existing scripts to run as-is.

As a reminder, we will also want to be sure we update all references to `wasp new foo` in the docs to supply a `-t basic` after them, as well as label this a breaking change IMO. 👍🏻

Reply:
I think this is a valid suggestion 👍 if they run `wasp new` only then it's interactive, if they do `wasp new <name>` it shouldn't go into the interactive mode